packages/engine/src/common/constants/Axis3D.ts

```diff
@@ -0,0 +1,44 @@
+// Common readonly 3D Axis definitions
+
+/** X Axis (1,0,0) */
+export const X = {
```

Comment:
We should be using standard types to ensure that functions remain highly optimized by the javascript JIT. 

This is what I've done for constants in the past:


```ts

export const V_00 = Object.freeze(new Vector2(0,0))
export const V_11 = Object.freeze(new Vector2(1,1))
export const V_000 = Object.freeze(new Vector3(0,0,0))
export const V_100 = Object.freeze(new Vector3(1,0,0))
export const V_010 = Object.freeze(new Vector3(0,1,0))
export const V_001 = Object.freeze(new Vector3(0,0,1))
export const V_111 = Object.freeze(new Vector3(1,1,1))
export const Q_IDENTITY = Object.freeze(new Quaternion)

export const DIRECTION = {
    RIGHT: Object.freeze(new Vector3(1, 0, 0)),
    UP: Object.freeze(new Vector3(0, 1, 0)),
    NEAR: Object.freeze(new Vector3(0, 0, 1)),
    LEFT: Object.freeze(new Vector3(-1, 0, 0)),
    DOWN: Object.freeze(new Vector3(0, -1, 0)),
    FAR: Object.freeze(new Vector3(0, 0, -1))
}
```